src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -209,33 +303,16 @@ private async Task WriteCsvHeaderRow(CsvWriter csv, PermalinkCsvMetaViewModel me
         }
 
         // TODO EES-3755 Remove after Permalink snapshot work is complete
-        private async Task<Either<ActionResult, LegacyPermalink>> FindLegacy(
-            Guid id,
+        private Task<Either<ActionResult, LegacyPermalink>> FindLegacy(
+            Guid permalinkId,
             CancellationToken cancellationToken)
         {
-            return await _contentDbContext.Permalinks
+            return _contentDbContext.Permalinks
```

Comment:
I don't think it's a question of safety, regardless of whether `await` is used here, the caller still has to await this method call to get the result otherwise they'd have either a Task that's not started or a Task that's started but might not have run to completion yet.

As discussed at our meeting there does seem to be some pro's and cons to each. Returning `Task` is more performant, while returning `async Task` is better for exception handling.

I'm reverting this change and other changes to return `async Task` as we've agreed to use that rather than returning `Task`.